### PR TITLE
Enable Access-Control-Allow-Origin header

### PR DESCRIPTION
**Pull request by valentinbootz**
# Access-Control-Allow-Origin Header

## :recycle: Current situation & Problem

[comment]: # "
*Describe the current situation (if possible with and exemplary (or real) code snippet and/or where this is used)*
*Please link any open issue that is addressed with this PR*
"
Currently there is no option to specify `Access-Control-Allow-Origin` headers in a `HTTPResponse`

## :bulb: Proposed solution

[comment]: # "
*Describe the solution and how this affects the project and internal structure*
"
This PR implements the `Access-Control-Allow-Origin` header field that enables users to indicate the response can be shared with requesting code from the given origin or use the wildcard to allow requesting code from any origin to access the resource.
```
app.httpServer.registerRoute(.GET, endpoint.httpPathComponents) { req in
	HTTPResponse(
		version: req.version,
		status: .ok,
		headers: HTTPHeaders {
			$0[.accessControlAllowOrigin] = .wildcard
		},
		bodyStorage: .buffer(initialValue: options.format.string(of: document))
	)
}
```

## :gear: Release Notes 

[comment]: # "
*Add a short summary of the feature as well as possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented if it appends or changes the public interface.*
"
- Add access-control-allow-origin http response header

## :heavy_plus_sign: Additional Information

[comment]: # "
*Provide some additional information if possible*
"

### Related PRs

[comment]: # "
*Reference the related PRs*
"

### Testing

[comment]: # "
*Are there tests included? If yes, which situations are tested and which corner cases are missing?*
"

### Reviewer Nudging

[comment]: # "
*Where should the reviewer start, where is a good entry point?*
"

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/Apodini/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/Apodini/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/Apodini/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/Apodini/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by valentinbootz** [comment]
Danke dir @lukaskollmer für deine Lösung! 🚀💪🏼 Sieht gut aus bei mir. Ich habe noch den access modifier der AnyHTTPHeaderName extension angepasst. Schau gerne nochmal rein falls du noch etwas an der Implementierung anpassen möchtest.

File: Sources/ApodiniNetworkingHTTPSupport/Headers.swift [hunk 1]
```diff
@@ -599,3 +599,37 @@ extension Array: HTTPHeaderFieldValueCodable where Element: HTTPHeaderFieldValue
             .joined(separator: ", ")
     }
 }
+
+public extension AnyHTTPHeaderName {
+    /// The `Access-Control-Allow-Origin` header field
```
Comment (lukaskollmer) [inline Sources/ApodiniNetworkingHTTPSupport/Headers.swift]: Can you move this to the `extension` we already have somewhere above that contains all of the other non-HTTP2-specific header names?

**Review by lukaskollmer — APPROVED** [review]
LGTM 👍. You'll probably need to add a test case or two to get the codecov thing to become green.

File: Sources/ApodiniNetworkingHTTPSupport/Headers.swift (hunk 1 above)
Comment (valentinbootz) [reply Sources/ApodiniNetworkingHTTPSupport/Headers.swift]: Yes good catch 💪
